apps/experiments/models.py

```diff
@@ -512,6 +512,39 @@ def get_experiments_for_display(self):
             .distinct()
         )
 
+    @transaction.atomic()
+    def update_memory(self, data: dict, experiment: Experiment | None = None):
+        """
+        Updates this participant's data records by merging `data` with the existing data. By default, data for all
+        experiments the this participant participated in will be updated. If there are no records for a specific
+        experiment, one will be created.
+
+        Paramters
+        data:
+            A dictionary containing the new data
+        experiment:
+            If specified, only the data for this experiment will be updated
+        """
+        if experiment:
+            experiments = [experiment]
+        else:
+            experiments = Experiment.objects.filter(team=self.team).prefetch_related(
+                Prefetch("participant_data", queryset=ParticipantData.objects.filter(participant=self))
+            )
+
+        records_to_update = []
+        for experiment in experiments:
+            participant_data = experiment.participant_data.first()
```

Comment:
I just realised that in the case of a single experiment being passed in, this will be data for all participants. I don't quite know how to handle that case except perhaps to have a getter function which is different for each case. Another option would be to separate this into two functions, one for global updates and one for experiment specific.

Reply:
+1 I was thinking splitting into 2 functions, but also the getter sounds good too as long as the naming is clear enough

Reply:
Ah, snappity snap you're right. Let me make an update